Automattic/newspack-newsletters
# Hotfix: Data reports: fix MC data reports

### All Submissions:






### Changes proposed in this Pull Request:



Sent/opens/clicks data in list activity endpoint (`/list/:id/activity`) from the MC API – as it turns out – is available after a 2-3 day delay. Reporting daily for the day before (as our use of the data reports is set up) thus would not work. This PR adjusts the usage report generation, so that the lists activity endpoint is used for subscribes and unsubscribes, but for sent/opens/clicks data, campaigns reports endpoint (`/reports`) is used. 

Because the latter endpoint returns per-campaign data (not per-day), last fetched data is saved and then subtracted from the new data. This way a picture of _daily_ sent/opens/clicks can be built. 

### How to test the changes in this Pull Request:

1. Ensure Mailchimp is configured as the ESP 
2. Run `wp eval "echo print_r( Newspack_Newsletters_Mailchimp_Usage_Reports::get_usage_report()->to_array(), true );"` and observe a usage report is printed

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207426954608290
